modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/BaseTemplateUpgradeProcessTestCase.java

```diff
@@ -0,0 +1,156 @@
+			_group.getGroupId(), TestPropsValues.getUserId());
+	}
+
+	protected void addDDMTemplate(String filePath) throws Exception {
+		long classPK = ParamUtil.getLong(_serviceContext, "classPK");
+
+		ddmTemplate = _ddmTemplateService.addTemplate(
+			_group.getGroupId(), _portal.getClassNameId(DDMTemplate.class),
+			classPK, _portal.getClassNameId(JournalArticle.class),
+			HashMapBuilder.put(
+				LocaleUtil.US, "DDMTemplate Name"
+			).build(),
+			HashMapBuilder.put(
+				LocaleUtil.US, "DDMTemplate Description"
+			).build(),
+			DDMTemplateConstants.TEMPLATE_TYPE_DISPLAY, StringPool.BLANK,
+			TemplateConstants.LANG_TYPE_FTL, read(filePath), _serviceContext);
+	}
+
+	protected void addFragmentEntry(String filePath) throws Exception {
+		FragmentCollection fragmentCollection =
+			_fragmentCollectionService.addFragmentCollection(
+				_group.getGroupId(), "Fragment Collection", StringPool.BLANK,
+				_serviceContext);
+
+		fragmentEntry = _fragmentEntryService.addFragmentEntry(
+			_group.getGroupId(), fragmentCollection.getFragmentCollectionId(),
+			null, "FragmentEntry Name", null, read(filePath), null, false, null,
+			null, 0, FragmentConstants.TYPE_COMPONENT, null,
+			WorkflowConstants.STATUS_APPROVED, _serviceContext);
+	}
+
+	protected DDMTemplate getDDMTemplate() throws Exception {
+		return _ddmTemplateService.getTemplate(ddmTemplate.getTemplateId());
+	}
+
+	protected FragmentEntry getFragmentEntry() throws Exception {
+		return _fragmentEntryService.fetchFragmentEntry(
+			fragmentEntry.getFragmentEntryId());
+	}
+
+	protected abstract String getUpgradeStepClassName() throws Exception;
+
+	protected String read(String filePath) throws Exception {
+		Class<?> clazz = getClass();
+
+		return StringUtil.read(
+			clazz.getClassLoader(),
+			"com/liferay/dynamic/data/mapping/dependencies/upgrade" + filePath);
+	}
+
+	protected void runTemplateUpgrade() throws Exception {
+		try (LogCapture logCapture = LoggerTestUtil.configureLog4JLogger(
+				getUpgradeStepClassName(), LoggerTestUtil.OFF)) {
+
+			UpgradeProcess upgradeProcess = UpgradeTestUtil.getUpgradeStep(
+				_upgradeStepRegistrator, getUpgradeStepClassName());
+
+			upgradeProcess.upgrade();
+
+			_multiVMPool.clear();
+		}
+	}
+
+	@Inject
+	protected DDMTemplate ddmTemplate;
+
+	@Inject
+	protected FragmentEntry fragmentEntry;
+
+	@Inject(
+		filter = "(&(component.name=com.liferay.dynamic.data.mapping.internal.upgrade.registry.DDMServiceUpgradeStepRegistrator))"
+	)
+	private static UpgradeStepRegistrator _upgradeStepRegistrator;
+
+	@Inject
+	private DDMTemplateService _ddmTemplateService;
+
+	@Inject
+	private FragmentCollectionService _fragmentCollectionService;
+
+	@Inject
+	private FragmentEntryService _fragmentEntryService;
+
+	@DeleteAfterTestRun
+	private Group _group;
+
+	@Inject
+	private MultiVMPool _multiVMPool;
+
+	@Inject
+	private Portal _portal;
+
+	@Inject
+	private ServiceContext _serviceContext;
```

Comment:
@alberttgomes, I don't think that we need to annotate this with `@Inject`, actually we shouldn't use any annotation. Can you please give it a try?

Reply:
Sure!